src/NHibernate.Test/NHSpecificTest/GH1496/Mappings.hbm.xml

```diff
@@ -0,0 +1,63 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<hibernate-mapping xmlns="urn:nhibernate-mapping-2.2" assembly="NHibernate.Test" namespace="NHibernate.Test.NHSpecificTest.GH1496">
+  <class name="Person" dynamic-update="true"   dynamic-insert="false" select-before-update="true" >
+    <id name="Id" type="integer" />
+    <property name="Name" type="string" length="50"/>
+    <many-to-one name="Address" class="Address" column="AddressId"/>
+  </class>
+
+  <class name="Address" dynamic-update="true" lazy="true">
+    <id name="Id" type="integer">
+      <generator class="native" />
+    </id>
+    <property name="PostalCode" type="string" length="20" />
+    <property name="Street" type="string" length="100" />
+    <property name="State" type="string" length="50" />
+  </class>
+
+  <class name="Employee" dynamic-update="true"   dynamic-insert="false" select-before-update="true" >
+    <id name="Id" type="integer" />
+    <property name="Name" type="string" length="50"/>
+    <many-to-one name="HomeContact" entity-name="HomeContact" cascade="persist,save-update,delete" fetch="join">
+      <column name="HomeContactId"/>
+      <formula>'HOME'</formula>
+    </many-to-one>
+    <many-to-one name="WorkContact" entity-name="WorkContact" cascade="persist,save-update,delete" fetch="join">
+      <column name="WorkContactId"/>
+      <formula>'WORK'</formula>
+    </many-to-one>
+  </class>
+
+  <class name="Contact"
+		table="Contact"
+		entity-name="HomeContact"
+		where="TypeName='HOME'"
+		check="TypeName in ('HOME', 'WORK')"
```

Comment:
This looks unsupported by Firebird. Is having this relevant for the bug? If no, please simplify as much as possible the test for having it as minimal as possible for demonstrating the bug.
Otherwise override the `AppliesTo(Dialect dialect)` in your test case, and add something like
```c#
if (dialect is FireBirdDialect)
    Assert.Ignore("Firebird is not supported by this test");
```

Reply:
Sure, I will simplify it.

Reply:
Also I will follow your instructions and create new branch.